AppleALC/ALCUserClientProvider/ALCUserClientProvider.hpp

```diff
@@ -0,0 +1,43 @@
+//
+//  ALCUserClientProvider.hpp
+//  AppleALC
+//
+//  Created by Nick on 10/14/20.
+//  Copyright © 2020 vit9696. All rights reserved.
+//
+
+#ifndef ALCUserClientProvider_hpp
+#define ALCUserClientProvider_hpp
+
+#define kIOHDACodecDevice	"IOHDACodecDevice"
+
+#include<IOKit/IOService.h>
+#include <IOKit/IOUserClient.h>
+
+#include "../kern_alc.hpp"
+
+class ALCUserClientProvider : public IOService {
+	typedef IOService super;
+	OSDeclareDefaultStructors(ALCUserClientProvider);
+	
+private:
+	IOService* 	mHDACodecDevice { nullptr };
+	bool		readyForVerbs	{ false	};
+	
+public:
+	virtual bool start(IOService* provider) override;
+	virtual void stop(IOService* provider) override;
+	
+	/**
+	 *	Called by user-client to set the codec verbs
```

Comment:
Something is up with the indentation here and and with the `@param` below. Should not it be indented with two spaces?

Reply:
Classic space vs tabs error. Xcode showed me proper indentation (visually) whereas it actually was indented with 4 spaces. Resolved.